src/CommentDialog.jsx

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable react/prop-types */
+import * as React from "react";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import { postComment } from "./utils";
+
+export default function CommentDIalog({ id, setNewComment }) {
+  const [open, setOpen] = React.useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <React.Fragment>
+      <Button
+        sx={{ margin: "1rem" }}
+        variant="contained"
+        onClick={handleClickOpen}
+      >
+        COMMENT
+      </Button>
+      <Dialog
+        maxWidth={"sm"}
+        fullWidth={true}
+        open={open}
+        onClose={handleClose}
+        PaperProps={{
+          component: "form",
+          autoComplete: "off",
+          onSubmit: async (event) => {
+            event.preventDefault();
+            const formData = new FormData(event.currentTarget);
```

Comment:
I would extract this out to its own function in the component to make debugging easier